spec/features/pages_visited_spec.rb

```diff
@@ -3,24 +3,29 @@
 RSpec.feature "pages visited" do
   scenario "displays pages visited" do
     visitor = FactoryBot.create(:visitor)
-    survey1 = FactoryBot.create(:survey, started_at: "2020-04-02 00:00:00", visitor: visitor)
-    survey2 = FactoryBot.create(:survey, started_at: "2020-04-03 00:00:00", visitor: visitor)
+    survey = FactoryBot.create(:survey, started_at: "2020-04-02 00:00:00", visitor: visitor)
 
-    @phrase = FactoryBot.create(:phrase, phrase_text: "how government works")
-    @survey_answer1 = FactoryBot.create(:survey_answer, survey: survey1, answer: "I want to understand how government works")
-    @survey_answer2 = FactoryBot.create(:survey_answer, survey: survey2, answer: "How government works is important")
+    phrase = FactoryBot.create(:phrase, phrase_text: "how government works")
+    survey_answer = FactoryBot.create(:survey_answer, survey: survey)
+    FactoryBot.create(:mention, phrase: phrase, survey_answer: survey_answer)
 
-    FactoryBot.create(:mention, phrase: @phrase, survey_answer: @survey_answer1)
-    FactoryBot.create(:mention, phrase: @phrase, survey_answer: @survey_answer2)
+    user_visit1 = FactoryBot.create(:visit, visitor: visitor)
+    user_visit2 = FactoryBot.create(:visit, visitor: visitor)
+    FactoryBot.create(:survey_visit, survey: survey, visit: user_visit1)
+    FactoryBot.create(:survey_visit, survey: survey, visit: user_visit2)
 
-    user_visit = FactoryBot.create(:visit, visitor: visitor)
-    pages = FactoryBot.create_list(:page, 20)
-    pages.each { |p| FactoryBot.create(:page_visit, page: p, visit: user_visit) }
+    create_pages_and_page_visits_for_visit(10, user_visit1)
+    create_pages_and_page_visits_for_visit(10, user_visit2)
 
-    visit pages_visited_path(@phrase)
+    visit pages_visited_path(phrase)
 
     expect(page).to have_link("/test-page-1")
     expect(page).to have_link("/test-page-15")
     expect(page).to have_content("Next page")
   end
 end
+
+def create_pages_and_page_visits_for_visit(number_of_pages, visit)
+  pages = FactoryBot.create_list(:page, number_of_pages)
+  pages.each { |p| FactoryBot.create(:page_visit, page: p, visit: visit) }
```

Comment:
super pedantic but I don't like having single character variable names, it's clearer and easier for other people if we rename `p` to `page` and increases the line length by only 6 characters

Reply:
Happy to rename it in this method. It's probably `p` after I extracted from the test itself, where if you specify `page` then you're referencing the Capybara `page` var instead.